src/Entity/Pap/BuildingBlock.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+namespace App\Entity\Pap;
+
+use ApiPlatform\Core\Annotation\ApiResource;
+use App\Entity\Adherent;
+use App\Entity\EntityTimestampableTrait;
+use App\Pap\BuildingStatusEnum;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+use Symfony\Component\Serializer\Annotation\Groups;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="pap_building_block")
+ *
+ * @ApiResource(
+ *     attributes={
+ *         "normalization_context": {
+ *             "groups": {"pap_building_block_list"},
+ *             "iri": true,
+ *         },
+ *         "pagination_enabled": false,
+ *     },
+ *     collectionOperations={},
+ *     itemOperations={},
+ * )
+ */
+class BuildingBlock
+{
+    use EntityTimestampableTrait;
+
+    /**
+     * @ORM\Column(type="integer", options={"unsigned": true})
+     * @ORM\Id
+     * @ORM\GeneratedValue
+     */
+    private ?int $id;
```

Comment:
`private ?int $id = null;`

otherwise you will have a fatal error if you call `$buildingBlock->getId();` on a new object